### DISCUSSION TITLE
How to know the state of a connection

### DISCUSSION THREAD

**Discussion by acbeni** [stephenh/ts-proto#470] [Q&A] [answered]
Hello there and thanks for the wonderful work.

I have a question, i have an application in which its beneficial to know the connection status (a grpc stream connection status to be more precise). the client generated from ts-node retries to reconnect which is good (though i would prefer to have more control over the retry strategy). but the issue i have is that i cant know if the server is down for example since the client keep retrying. and thus i cant print a proper error message for the client

If there is a way to get hold of a stream status, like is is `open` `closed` `retrying`....?

Thanks

**Comment by stephenh** [comment, marked as answer]
Super-late follow up @acbeni , but ts-proto is generally more of a low-level "roll your own RPC as-needed" infra tool and I'd defer answering this to the higher-level / out-of-the-box  framework like @aikoven 's nice-grpc:

https://github.com/deeplay-io/nice-grpc

(...which I've not used, but looks really well done, so I am naively assuming it has a way to do this. :-))

**Reply by acbeni** [reply]
Thank you very much @stephenh for the recommendition, never came accross [nice-grpc](https://github.com/deeplay-io/nice-grpc) wile searching. and i really would love to thank you again for this open source project. it made my life much easier <3
